src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeonardoModel.scala

```diff
@@ -100,6 +103,31 @@ object LeonardoJsonSupport extends SprayJsonSupport with DefaultJsonProtocol {
     }
   }
 
+  implicit object URLFormat extends JsonFormat[URL] {
+    def write(obj: URL) = JsString(obj.toString)
+
+    def read(json: JsValue): URL = json match {
+      case JsString(url) => new URL(url)
+      case other => throw DeserializationException("Expected URL, got: " + other)
+    }
+  }
+
+  case class TypedStringFormat[T <: TypedString](create: String => T) extends JsonFormat[T] {
+    def write(obj: T): JsValue = JsString(obj.toString)
+
+    def read(json: JsValue): T = json match {
+      case JsString(value) => create(value)
```

Comment:
A function from `String` to a specific subclass of `TypedString`, e.g. the implicit `GoogleProject.apply()` created by virtue of it being a case class